src/main/scala/shell/UARTOverlay.scala

```diff
@@ -12,7 +12,7 @@ import freechips.rocketchip.interrupts.IntInwardNode
 
 //dont make the controller here
 //move flowcontrol to shell input?? 
-case class UARTShellInput()
+case class UARTShellInput(number: Int = 0)
```

Comment:
I'd maybe call this `index` or `id` or something? To me `number` sounds like a total count not a particular one-of-many

Reply:
Yeah idk why i settled on number, index is much better